kvbc/src/ReplicaImp.cpp

```diff
@@ -232,7 +238,8 @@ ReplicaImp::~ReplicaImp() {
 
 Status ReplicaImp::addBlockInternal(const SetOfKeyValuePairs &updates, BlockId &outBlockId) {
   m_lastBlock++;
-  m_appState->m_lastReachableBlock++;
+  auto pt = downcast_appstate();
```

Comment:
If you have an RoRAppState, this downcast would fail, right?

Reply:
This function is called only with non-ro replica, but you are right -  I will think of how to make bit more clear and safe.